source/locationHelper.py

```diff
@@ -0,0 +1,402 @@
+		if not isinstance(other,_RECT_CLASSES):
+			try:
+				other=toRect(other)
+			except ValueError:
+				return False
+		return other.left >= self.left and other.top >= self.top and other.right <= self.right and other.bottom <= self.bottom
+
+	def __eq__(self, other):
+		if not isinstance(other,_RECT_CLASSES):
+			try:
+				other=toRect(other)
+			except ValueError:
+				return False
+		return other.left == self.left and other.top == self.top and other.right == self.right and other.bottom == self.bottom
+
+	def __neq__(self, other):
+		if not isinstance(other,_RECT_CLASSES):
+			try:
+				other=toRect(other)
+			except ValueError:
+				return False
+		return not (other.left == self.left and other.top == self.top and other.right == self.right and other.bottom == self.bottom)
+
+	def __sub__(self,other):
+		if not isinstance(other,_RECT_CLASSES):
+			return NotImplemented
+		left,top,right,bottom=self.left-other.left,self.top-other.top,self.right-other.right,self.bottom-other.bottom
+		if type(self) is Location:
+			return Location(left,top,right-left,bottom-top)
+		return Rect(left,top,right,bottom)
+
+	def __rsub__(self,other):
+		return self.__sub__(other)
+
+	def __and__(self,other):
+		"""Returns the intersection of self and other.
+		For example, if self = Rect(left=10,top=10,right=25,bottom=25) and other = Rect(left=20,top=20,right=35,bottom=35),
+		this results in Rect(left=20,top=20,right=25,bottom=25).
+		No intersect results in a rectangle with zeroed coordinates.
+		"""
+		if not isinstance(other,_RECT_CLASSES):
+			try:
+				other=toRect(other)
+			except ValueError:
+				return NotImplemented
+		left=max(self.left,other.left)
+		top=max(self.top,other.top)
+		right=min(self.right,other.right)
+		bottom=min(self.bottom,other.bottom)
+		if left>right or top>bottom:
+			left=top=right=bottom=0
+		if type(self) is Location:
+			return Location(left,top,right-left,bottom-top)
+		return Rect(left,top,right,bottom)
+
+	def __rand__(self,other):
+		return self.__and__(other)
+
+class Location(_RectMixin, namedtuple("Location",("left","top","width","height"))):
+	"""Represents a rectangle on the screen, based on left and top coordinates, width and height."""
+
+	@property
+	def right(self):
+		return self.left+self.width
+
+	@property
+	def bottom(self):
+		return self.top+self.height
+
+	def toRect(self):
+		return Rect(self.left,self.top,self.right,self.bottom)
+
+class Rect(_RectMixin, namedtuple("Rect",("left","top","right","bottom"))):
+	"""Represents a rectangle on the screen.
+	By convention, the right and bottom edges of the rectangle are normally considered exclusive.
+	"""
+
+	@property
+	def width(self):
+		return self.right-self.left
+
+	@property
+	def height(self):
+		return self.bottom-self.top
+
+	def toLocation(self):
+		return Location(self.left,self.top,self.width,self.height)
+
+def toRect(*params):
+	"""
+	Converts the given input to L{Rect}.
+	Input should be one of the following types:
+		* One of l{_RECT_CLASSES}.

```

Comment:
Judging by the underscore, this is a private variable, which should probably not be in your documentation.

Reply:
This part of the docs should definitely be public, so I'll also make these variables public.